common/events_io_test.cpp

```diff
@@ -0,0 +1,386 @@
+// Copyright 2023 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+#include <common/events_io.hpp>
+
+#include <vector>
+#include <fstream>
+
+#include <gtest/gtest.h>
+
+#include <common/setup.hpp>
+#include <common/testing.hpp>
+
+using namespace std;
+
+namespace error = mender::common::error;
+namespace events = mender::common::events;
+namespace io = events::io;
+namespace mtesting = mender::common::testing;
+
+using TestEventLoop = mtesting::TestEventLoop;
+
+class EventsIo : public testing::Test {
+protected:
+	void SetUp() override {
+		mender::common::setup::GlobalSetup();
+	}
+};
+
+TEST_F(EventsIo, ReadAndWriteWithPipes) {
+	TestEventLoop loop;
```

Comment:
Q: Isn't this really more like: `timeout` or similar?

Reply:
Well, kind of, but its primary purpose is still to be an event loop, the timeout is just a fallback. Wouldn't it look funny with `timeout.Run()` to run an event loop?